### PR TITLE
Add ability to list by tags and/or ward

### LINKED ISSUE #36: As a ward nurse, I can list all patients tied to a specific ward
so that I know which patients belong to which ward.

### LINKED ISSUE #40: As a ward nurse, I can list patients based on their tags
so that I can search patients based on category.

### PR DESCRIPTION
**Pull request by ryanlimdx**
Update list to accept parameters to filter list of patients by their tags and/or ward, along with testcases.

### CODE REVIEW

**Review by ejnan — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 1]
```diff
@@ -12,13 +16,57 @@ public class ListCommand extends Command {
 
     public static final String COMMAND_WORD = "list";
 
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Lists relevant contacts from the address book. "
+            + "Parameters: "
+            + "[" + PREFIX_TAG + "TAG]..."
+            + "[" + PREFIX_WARD + "WARD]\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_TAG + "Fall-risk "
```
Comment (ejnan) [inline src/main/java/seedu/address/logic/commands/ListCommand.java]: Should we decide whether we want to stick to FallRisk or Fall-risk?

**Review by ejnan — COMMENTED** [review]
LGTM, just some nits as we should standardise input format for tags, since it only accepts alphanumeric values. Should we stick to Fall-risk or FalRisk?

File: src/main/java/seedu/address/logic/commands/ListCommand.java (hunk 1 above)
Comment (ryanlimdx) [reply src/main/java/seedu/address/logic/commands/ListCommand.java]: I will change it to FallRisk. Thanks!

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 2]
```diff
@@ -85,7 +85,6 @@ public void parseCommand_help() throws Exception {
     @Test
     public void parseCommand_list() throws Exception {
         assertTrue(parser.parseCommand(ListCommand.COMMAND_WORD) instanceof ListCommand);
-        assertTrue(parser.parseCommand(ListCommand.COMMAND_WORD + " 3") instanceof ListCommand);
```
Comment (iamtr) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:88]: If this expected behavior is false, would it be better to add another assertFalse statement?

**Review by iamtr — COMMENTED** [review]
LGTM, just one minor issue with the test case

**Pull request merged**